llvm/test/Feature/Repo/repo_extern_weak_symbol.ll

```diff
@@ -0,0 +1,38 @@
+; Check the weak symbol binding for the following 4 cases:
+;   Case 1: when the symbol is declared as extern_weak function `h()`, it is referenced by global variable `c`.
+;   Case 2: when the symbol is declared as extern_weak function `h()`, it is referenced by function `g`.
+;   Case 3: when the symbol is declared as extern_weak variable `a`, it is referenced by global variable `b`.
+;   Case 4: when the symbol is declared as extern_weak variable `a`, it is referenced by function `use-a`.
+;
+; RUN: rm -f %t.db
+; RUN: env REPOFILE=%t.db opt -S %s -o %t.ll
+; RUN: env REPOFILE=%t.db llc -filetype=obj %t.ll -o %t.o
+; RUN: pstore-dump --all-fragments %t.db | FileCheck %s
```

Comment:
Isn't usual for the compiler tests to work by inspecting the debugging out (c.f. repo_ticket_file_digest.ll) rather than by running pstore-dump? This way we minimise dependency on a particular database back-end.

Reply:
Good point. I have updated the patch following your suggestion.